### PR TITLE
Add ConfirmParentModule

### LINKED ISSUE #123: Mark as community consensus if linked > 0 and unconfirmed
**What is the problem the module is aiming to solve? Please describe.**
Mark as community consensus if linked > 0 and unconfirmed

### PR DESCRIPTION
**Pull request by SPGoding**
## Purpose
Resolve #123

## Approach
Added a new ConfirmParentModule which will set the Confirmation Status of tickets whose Linked is greater than `linkedThreshold` and Confirmation Status exists in `confirmationStatusWhitelist` to the value specified in `targetConfirmationStatus`.

#### Checklist
- [x] Included tests
- [ ] Tested in MCTEST-xxx

### CODE REVIEW

**Comment by SPGoding** [comment]
I just got some new ideas. Let's do this latter.

File: src/main/kotlin/io/github/mojira/arisa/modules/ConfirmParentModule.kt [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+    override fun invoke(request: ConfirmParentModule.Request): Either<ModuleError, ModuleResponse> = with(request) {
+        Either.fx {
+            assertLinkedMoreThanZero(linked).bind()
+            assertConfirmationStatusWhitelisted(confirmationStatus, confirmationStatusWhitelist).bind()
+            setConfirmationStatus(targetConfirmationStatus).toFailedModuleEither().bind()
+        }
+    }
+
+    private fun assertLinkedMoreThanZero(linked: Double?) = if ((linked ?: 0.0) > 0.0) {
```
Comment (urielsalis) [inline src/main/kotlin/io/github/mojira/arisa/modules/ConfirmParentModule.kt:26]: I would change it to more than 3 to really make sure

File: arisa.json [hunk 2]
```diff
@@ -109,6 +109,18 @@
           "MCPE"
         ]
       },
+      "confirmParent": {
+        "confirmationStatusWhitelist": ["Unconfirmed", "Plausible"],
+        "targetConfirmationStatus": ["Community Consensus"],
+        "resolutions": [
+          "Unresolved"
+        ],
```
Comment (NeunEinser) [inline arisa.json:117]: unresolved is default, no need to specify this when it's only unresolved.

File: src/test/kotlin/io/github/mojira/arisa/modules/ConfirmParentModuleTest.kt [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+            Unit.right()
+        }
+
+        val result = module(request)
+
+        result.shouldBeRight(ModuleResponse)
+        changedConfirmation.shouldBe("Community Consensus")
+    }
+
+    "should return OperationNotNeededModuleResponse when Confirmation Status is Community Consensus and Linked is 1" {
```
Comment (NeunEinser) [inline src/test/kotlin/io/github/mojira/arisa/modules/ConfirmParentModuleTest.kt:73]: I would move these up to where other OperationNotNeededModuleResponse are, to match the order of other modules

File: src/test/kotlin/io/github/mojira/arisa/modules/ConfirmParentModuleTest.kt [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        val result = module(request)
+
+        result.shouldBeLeft(OperationNotNeededModuleResponse)
+    }
+
+    "should set to Community Consensus when Confirmation Status is unset and Linked is 1" {
+        var changedConfirmation = ""
+
+        val module = ConfirmParentModule(listOf("Unconfirmed", "Plausible"), "Community Consensus")
+        val request = ConfirmParentModule.Request(null, 1.0) {
```
Comment (NeunEinser) [inline src/test/kotlin/io/github/mojira/arisa/modules/ConfirmParentModuleTest.kt:32]: It's also sometimes the empty string, not sure about the exact conditions, but empty string should also be counted as "Unconfirmed"

File: src/main/kotlin/io/github/mojira/arisa/modules/ConfirmParentModule.kt (hunk 1 above)
Comment (SPGoding) [reply src/main/kotlin/io/github/mojira/arisa/modules/ConfirmParentModule.kt]: Added a new `linkedThreshold` configuration

File: src/main/kotlin/io/github/mojira/arisa/modules/ConfirmParentModule.kt [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+    override fun invoke(request: ConfirmParentModule.Request): Either<ModuleError, ModuleResponse> = with(request) {
+        Either.fx {
+            assertLinkedMoreThanZero(linked).bind()
+            assertConfirmationStatusWhitelisted(confirmationStatus, confirmationStatusWhitelist).bind()
+            setConfirmationStatus(targetConfirmationStatus).toFailedModuleEither().bind()
+        }
+    }
+
+    private fun assertLinkedMoreThanZero(linked: Double?) = if ((linked ?: 0.0) >= linkedThreshold) {
```
Comment (NeunEinser) [inline src/main/kotlin/io/github/mojira/arisa/modules/ConfirmParentModule.kt:27]: You probably should rename that function to reflect the new behavior.

Comment (SPGoding) [reply src/main/kotlin/io/github/mojira/arisa/modules/ConfirmParentModule.kt:27]: Oops, missed that

**Pull request merged**